File: backend/src/models/reading-session.js
const mongoose = require("mongoose");

const ReadingSessionSchema = new mongoose.Schema(
  {
    userId: { type: mongoose.Schema.ObjectId, ref: "User", required: true },
    bookId: { type: mongoose.Schema.ObjectId, ref: "Book", required: true },
    currentPage: { type: Number, default: 0 },
    totalPages: { type: Number, required: true },
  },
  { timestamps: true }
);

const ReadingSession = mongoose.model("ReadingSession", ReadingSessionSchema);

module.exports = ReadingSession;

File: frontend/lib/features/book/data/models/reading_session.dart
// ignore_for_file: invalid_annotation_target
import 'package:bookdf/features/book/data/models/book.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
part 'reading_session.freezed.dart';
part 'reading_session.g.dart';

@freezed
class ReadingSession with _$ReadingSession {
  factory ReadingSession({
    @JsonKey(name: '_id') required String id,
    required String userId,
    required String bookId,
    required int currentPage,
    required int totalPages,
    required Book bookDetails,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _ReadingSession;

  factory ReadingSession.fromJson(Map<String, dynamic> json) =>
      _$ReadingSessionFromJson(json);
}
